cmake/Modules/GrPython.cmake

```diff
@@ -146,12 +146,22 @@ if(NOT DEFINED GR_PYTHON_DIR)
 execute_process(
     COMMAND ${PYTHON_EXECUTABLE} -c "import os
 import site
-from distutils.sysconfig import get_python_lib
+import sys
+import sysconfig
 
 prefix = '${CMAKE_INSTALL_PREFIX}'
 
-#ask distutils where to install the python module
-install_dir = get_python_lib(plat_specific=True, prefix=prefix)
+#find where to install the python module
+if prefix is None:
+    prefix = os.path.normpath(sys.exec_prefix)
+# To do: How to handle on OSX or Windows
+if os.name == 'posix':
+    libdir = getattr(sys, 'platlibdir', 'lib')
+else :
+    libdir = 'lib'
+
+libpython = os.path.join(prefix, libdir,'python' + sysconfig.get_python_version())
+install_dir = os.path.join(libpython, 'site-packages')
```

Comment:
So this is the tricky part that needs to be replaced. `distutils`' `get_python_lib` returns `dist-packages` under some circumstances. e.g.
```python
>>> from distutils import sysconfig
>>> sysconfig.get_python_lib()
'/usr/lib/python3/dist-packages' 
>>> sysconfig.get_python_lib(plat_specific=True, prefix='/foo/bar')
'/foo/bar/lib/python3.8/site-packages'
```
Now, this is hard-coded to `site-packages`. I assume there should be at least an option for package maintainers to change that behavior.
Since this whole thing is kinda broken, I would appreciate more options anyways. e.g. an install with the default prefix `/usr/local` won't work "out-of-the-box" because the Python packages end up in `/usr/local/lib/python3.8/site-packages`, which is not in the `PYTHONPATH`.
```python
>>> import sys
>>> sys.path
['', '/usr/bin', '/usr/lib/python38.zip', '/usr/lib/python3.8', '/usr/lib/python3.8/lib-dynload', '/usr/local/lib/python3.8/dist-packages', '/usr/lib/python3/dist-packages']
```

Reply:
Seems like rather than hardcoding site-packages or using distutils, sysconfig should be able to be used here

Reply:
I found a solution completely based on sysconfig and it should work for different os.